packages/doenetml-worker-rust/doenetml-core/src/component/text.rs

```diff
@@ -1,20 +1,277 @@
+use std::cell::RefCell;
 use std::collections::HashMap;
+use std::rc::Rc;
 
-use crate::dast::Position as DastPosition;
+use doenetml_derive::add_standard_component_fields;
+
+use crate::component::ComponentProfile;
+use crate::dast::{
+    ElementData, FlatDastElement, FlatDastElementContent, FlatDastElementUpdate,
+    Position as DastPosition,
+};
+use crate::dependency::{Dependency, DependencyInstruction, DependencyValueUpdateRequest};
+use crate::state::{
+    StateVar, StateVarInterface, StateVarMutableViewTyped, StateVarParameters,
+    StateVarReadOnlyView, StateVarReadOnlyViewTyped, StateVarTyped, StateVarValue,
+};
+use crate::utils::KeyValueIgnoreCase;
 use crate::{ComponentChild, ComponentIdx, ExtendSource};
 
-use super::ComponentNode;
+use super::{
+    ComponentEnum, ComponentNode, ComponentNodeStateVariables, ComponentProfileStateVariable,
+    RenderedComponentNode,
+};
 
+#[add_standard_component_fields]
```

Comment:
This is nice, but as you say, less explicit. I was thinking something like,
```rust
struct ComponentCommonData {
  idx: ComponentIdx,
  parent: Option<ComponentIdx>,
  ...
}

struct Text {
  common: ComponentCommonData,
  render_data: TextRendererData,
  ...
}
```

Won't that do basically the same thing (possibly affecting the layout of the data structure, but I assume we don't care)?

Reply:
I could go either way. I feel like the way you suggest will cause fewer headaches for developers, as it is clearer what's going on. It could be confusing if a developer tries to add a field that conflicts with one of the standard fields and wonder where the conflict came from.

It was fun to learn to write an attribute macro, though!

Reply:
I removed the attribute macro in favor of this common struct.